ospfd/ospf_lsa.c

```diff
@@ -3648,3 +3648,34 @@ int ospf_lsa_refresh_walker(struct thread *t)
 
 	return 0;
 }
+
+/* Flush the LSAs for the specific area */
+void ospf_flush_lsa_from_area(struct ospf *ospf, struct in_addr area_id,
+			      int type)
+{
+	struct ospf_area *area;
+	struct route_node *rn;
+	struct ospf_lsa *lsa;
+
+	area = ospf_area_get(ospf, area_id);
+
+	switch (type) {
+	case OSPF_AS_EXTERNAL_LSA:
+		if ((area->external_routing == OSPF_AREA_NSSA) ||
+		    (area->external_routing == OSPF_AREA_STUB)) {
+			LSDB_LOOP (EXTERNAL_LSDB(ospf), rn, lsa)
+				if (IS_LSA_SELF(lsa) &&
+				    !(CHECK_FLAG(lsa->flags,
+						 OSPF_LSA_LOCAL_XLT)))
```

Comment:
why translator check is required here? you can't have NSSA and non-NSSA for the same area. Also this check is under stub check.

Reply:
In the nssa ABR case, any previously originated external LSAs need to be flushed. The check is only to skip the translated LSAs in the database. The same function is also called for stub area,  having the check does not cause any issue.